proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+                        "minLength": MIN_MODULE_NAME_LENGTH,
+                        "maxLength": MAX_MODULE_NAME_LENGTH,
+                        "pattern": "^[a-zA-Z0-9]*$]",
+                        "fieldNumber": 3
+                    }
+                }
+            }
+        },
+        "escrowSubstore": {
+            "type": "array",
+            "fieldNumber": 3,
+            "items": {
+                "type": "object",
+                "required": [
+                    "escrowedChainID",
+                    "nftID",
+                    "escrow"
+                ],
+                "properties": {
+                    "escrowedChainID": {
+                        "dataType": "bytes",
+                        "length": CHAIN_ID_LENGTH,
+                        "fieldNumber": 1
+                    },
+                    "nftID": {
+                        "dataType": "bytes",
+                        "length": NFT_ID_LENGTH,
+                        "fieldNumber": 2
+                    },
+                    "escrow": {
+                        "dataType": "boolean",
+                        "fieldNumber": 3
+                    }
+                }
+            }
+        },
+        "supportedNFTsSubstore": {
+            "type": "array",
+            "fieldNumber": 4,
+            "items": {
+                "type": "object",
+                "required": [
+                    "chainID",
+                    "supportedNftIDArray"
+                ],
+                "properties": {
+                    "chainID": {
+                        "dataType": "bytes",
+                        "length": CHAIN_ID_LENGTH,
+                        "fieldNumber": 1
+                    },
+                    "supportedNftIDArray": {
+                        "type": "array",
+                        "fieldNumber": 2,
+                        "items": {
+                            "dataType": "bytes",
+                            "length": NFT_ID_LENGTH
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
 ```
 
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the NFT module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisNFTStoreSchema` schema, given above.
+
+- Initial checks on the properties of `genesisBlockAssetObject`:
+
+  - Across all elements of the `NFTSubstore` array, all values given for `nftID` must be unique.
+  - For all elements of the `NFTSubstore` array, values given for `owner` must have either length `ADDRESS_ID_LENGTH` bytes (representing a user address) or `CHAIN_ID_LENGTH` bytes (representing a chain ID).
+  - The `NFTSubstore` must be in lexicographical order of `owner`.
```

Comment:
maybe this is a remnant of a previous version, but i think it should be in lexicographical order of `nftID`.